app/controllers/api/v1/products_controller.rb

```diff
@@ -37,6 +37,15 @@ def destroy
     head :no_content
   end
 
+  def transfer
+    return render_error(4209) unless new_owner
+
+    status = TransferProduct.new(product: product, new_owner: new_owner, current_owner: current_user).call
```

Comment:
you can use an instance variable to create the service, so you do not create it twice in the case you could use it more than once.
def transfer_product_service
  @... ||= TransferProduct.new......
end

Reply:
Also you don't need to  create the instance. It can be done y the class method. You would get the service instance after the call was executed and therefore, u would be able to ask for success? or error? methods.